server/util/grpc_client/grpc_client.go

```diff
@@ -62,11 +65,37 @@ func (p *ClientConnPool) Check(ctx context.Context) error {
 		}
 	}
 	if goodConns == 0 {
+		logConnPoolState(p.targetForLogging, p.conns)
 		return status.UnavailableError("No ready connections in gRPC connection pool")
 	}
 	return nil
 }
 
+func logConnPoolState(target string, conns []*clientConn) {
+	states := map[string]int{}
+	for _, c := range conns {
+		connState := c.GetState().String()
+		if _, found := states[connState]; !found {
```

Comment:
this is unnecessary for a map with primitive values
states[connState]++ will work even if the key is not present